homeassistant/components/dhcp/__init__.py

```diff
@@ -55,13 +58,33 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class DhcpServiceInfo(TypedDict):
+@dataclass
+class DhcpServiceInfo(BaseServiceInfo):
     """Prepared info from dhcp entries."""
 
-    ip: str
+    ip: str  # pylint: disable=invalid-name
     hostname: str
     macaddress: str
 
+    # Used to prevent log flooding. To be removed in 2022.6
+    _warning_logged: bool = False
+
+    def __getitem__(self, name: str) -> Any:
```

Comment:
Should this be excluded from coverage?

```suggestion
    def __getitem__(self, name: str) -> Any: # pragma: no cover
```

Reply:
I wouldn't do that. We want to remove it. The change to the project total coverage is insignificant.